content/en/tracing/trace_collection/automatic_instrumentation/single-step-apm/kubernetes.md

```diff
@@ -30,36 +30,109 @@ In a Kubernetes environment, use Single Step Instrumentation (SSI) for APM to in
 
-**Note:** To instrument only specific namespaces or pods, see [Advanced options](#advanced-options).
+**Note:** To instrument only specific namespaces or pods, seeworkload targeting in[Advanced options](#advanced-options).
 
 1. In Datadog, go to the [Install the Datadog Agent on Kubernetes][11] page.
 1. Follow the on-screen instructions to choose your installation method, select an API key, and set up the Operator or Helm repository.
 1. In the **Configure `datadog-agent.yaml`** section, go to **Additional configuration** > **Application Observability**, and turn on **APM Instrumentation**.
-   
+
    {{< img src="tracing/trace_collection/k8s-apm-instrumentation-toggle.jpg" alt="The configuration block for installing the Datadog Agent on Kubernetes through the Datadog app" style="width:100%;" >}}
 
 1. Deploy the Agent using the generated configuration file.
 1. Restart your applications.
 
 ## Configure Unified Service Tags
 
-Unified Service Tags (USTs) apply consistent tags across traces, metrics, and logs, making it easier to navigate and correlate your observability data. Learn how to [set USTs for Kubernetes services][5].
+Unified Service Tags (USTs) apply consistent tags across traces, metrics, and logs, making it easier to navigate and correlate your observability data.
+
+### Prerequisites
+
+To use the recommended ddTraceConfigs approach, ensure you have:
+
+| Component | Minimum Version | Notes |
+|-----------|-----------------|-------|
+| datadog-agent | 7.66+ | Required for valueFrom support |
+| datadog-operator | 1.16.0+ | 1.13.0+ works with agent version override |
+| datadog-helm-chart | 3.120.0+ | Added valueFrom support |
+
+### Recommended: Configure USTs with ddTraceConfigs
+
+With SSI, you can automatically extract UST values from pod labels and metadata without modifying individual deployments:
+
+```yaml
+datadog:
+  # Make pod labels available as tags in Datadog
+  kubernetesResourcesLabelsAsTags:
+    pods:
+      app-name: service
+    deployments.apps:
+      app-name: service
+    replicasets.apps:
+      app-name: service
+
+  # Set environment globally for the entire cluster
+  tags:
+    - "env:production"
 
-**Note**: USTs must be set on both the Deployment object and the Pod template spec.
+  apm:
+    instrumentation:
+      enabled: true
+      targets:
+        - name: my-services
+          podSelector:
+            matchExpressions:
+            - key: app-name           # Target pods with this label
+              operator: Exists
+          ddTraceConfigs:
+            - name: DD_SERVICE      # Extract service name from pod label
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.labels['app-name']
+            # DD_VERSION automatically extracted from image tags
+            # DD_ENV inherited from cluster-level tags above
+```
+
+**For namespace-based environments**, create separate targets:
```

Comment:
I would drop the namespace based environments for now, it's kind of confusing I think and the pod/deployment doesn't get the UST propagated, this only works well for service at the moment.